cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -811,11 +818,13 @@ class GroupByNode : public ExecNode, public TracedNode {
 
     auto handler = [this](const ExecBatch& full_batch, const Segment& segment) {
       if (!segment.extends && segment.offset == 0) RETURN_NOT_OK(OutputResult(false));
+      // This is not zero copy - we should refactor the code to pass
+      // offset and length to Consume to avoid copying here
       auto exec_batch = full_batch.Slice(segment.offset, segment.length);
       auto batch = ExecSpan(exec_batch);
```

Comment:
You're right this is not zero-copy, but I believe this can be fixed within this block of code, without changing the API. I'll accept this and will try to fix right after.

Reply:
Thank you

Reply:
I changed my post above because the fix I have in mind involves adding `ExecSpan::Slice`, which would require a bit of testing that I'd prefer to defer on.

Reply:
Sounds good